src/preamble.js

```diff
@@ -578,6 +578,23 @@ addOnPreRun(reportUndefinedSymbols);
 #endif
 #endif
 
+#if STACK_OVERFLOW_CHECK
+function stackCheckInit() {
+  // This is normally called automatically during __wasm_call_ctors but need to
+  // get these values before even running any of the ctors so we call it redundantly
+  // here.
+  // TODO(sbc): Move writeStackCookie to native to to avoid this.
```

Comment:
is this comment, and this TODO, still relevant? I think it isn't, as there is nothing redundant at this point - we call `stackCheckInit` once during preRun?

Reply:
I don't think I changed anything to make this comment less relevant.  I think the first comment is about _emscripten_stack_init/_emscripten_stack_set_limits.     The redundant part sounds like its referring to JS calling this and then native ctor calling it later.

Reply:
Indeed the redundancy is with system/lib/compiler-rt/stack_limits.S which adds emscripten_stack_init to the ctor list.  I'll improve the comment.